### ISSUE TITLE
Spring boot starter in another starter not exposed to compileClasspath

### ISSUE THREAD

**Issue by KRonst** [rubenlagus/TelegramBots#825]
Hello.
I'm using your [telegrambots-spring-boot-starter](https://github.com/rubenlagus/TelegramBots/tree/master/telegrambots-spring-boot-starter) in my own spring boot starter. Project is building with gradle 6.6.1.
In my starter I've add your starter in `implementation` configuration. And it's all ok. But when I add my starter to my another application also in `implementation` configuration, your starter and all it's libraries doesn't appear in `compileClasspath` of my application. So I can't use telegram classes in my application.
If I add your starter to my in `compile` configuration all works fine. But compile configuration is **deprecated** now.

**Comment by Chase22** [comment]
This is not a problem with the starter. The implementation configuration causes a dependency to not be part of the public api of a project, aka. it will not be included into the classpath of a project using your project as a dependency. This was introduced with the split of implementation and api dependencies, after depracating compilation dependencies.

You have to add the starter as and api dependency in order for it and all it's dependencies to also become direct dependencies of the new project.

see: https://docs.gradle.org/current/userguide/dependency_management_for_java_projects.html#sec:configurations_java_tutorial

**Comment by KRonst** [comment]
Yes, I've also tried to use `api` configuration to add telegrambots-spring-boot-starter into my starter. But it didn't help.

**Comment by leodev87** [comment]
Why don't you use normal library imports without spring-starter?

`@PropertySource("classpath:telegram-bots.properties")
@Slf4j
@Configuration
public class TelegramConfig {

    @Bean
    public TelegramBotsApi initTelegramBots(LDEAgentBot ldeAgentBot) {
        TelegramBotsApi botsApi = null;
        log.info("Begin Telegram bots registration");
        try {
            botsApi = new TelegramBotsApi(DefaultBotSession.class);
            botsApi.registerBot(ldeAgentBot);
        } catch (TelegramApiException e) {
            log.warn(e.getLocalizedMessage());
        }
        log.info("{} init ok...", TelegramBotsApi.class.getSimpleName());
        return botsApi;
    }
}`


`@Slf4j
@Component
public class MyBot extends AbilityBot {
//...
}`
